pkg/sql/opt/optbuilder/create_function.go

```diff
@@ -47,6 +52,25 @@ func (b *Builder) buildCreateFunction(cf *tree.CreateFunction, inScope *scope) (
 		b.schemaDeps = nil
 		b.schemaTypeDeps = util.FastIntSet{}
 		b.qualifyDataSourceNamesInAST = false
+
+		b.semaCtx.FunctionResolver = preFuncResolver
+		switch recErr := recover().(type) {
```

Comment:
extract a helper give you have this twice

Reply:
good point, done.